src/almanack/metrics/data.py

```diff
@@ -770,3 +776,43 @@ def get_github_build_metrics(
 
     # else we return an empty dictionary
     return {}
+
+
+def get_ecosystems_package_metrics(repo_url: str) -> Dict[str, Any]:
```

Comment:
Correct me if I'm wrong, but I think your intent with these methods is to return correctly-shaped data rather than raising an exception, even in cases when the actual data couldn't be retrieved. While I think that makes sense for a reporting component that should try to get as much data as possible rather than crashing, I could imagine someone running the tool and being surprised to find that the results from this method are all 0, and it would be difficult to debug why with the current architecture.

I'd suggest adding some warning+-level logging so that, if a user did want to investigate why they got the stats back that they did, they'd have somewhere to start. For example, the `get_api_data()` call below could fail because the remote URL isn't in the right format, and without any reporting either in `get_api_data()` or here it'd be difficult to determine that that's the source of the issue.

Reply:
Thanks, great point! I'll add in a warning message bout this.